File: src/main/java/org/grimrose/sqlap22/domain/java/Bug.java
package org.grimrose.sqlap22.domain.java;


import com.google.common.base.Optional;

import java.util.Date;

public class Bug {

    long bugId;
    Date dataReported;
    Optional<String> summary;
    long reportedBy;
    Optional<Long> assignedTo;
    BugStatus status;

    public long getBugId() {
        return bugId;
    }

    public void setBugId(long bugId) {
        this.bugId = bugId;
    }

    public Date getDataReported() {
        return dataReported;
    }

    public void setDataReported(Date dataReported) {
        this.dataReported = dataReported;
    }

    public Optional<String> getSummary() {
        return summary;
    }

    public void setSummary(Optional<String> summary) {
        this.summary = summary;
    }

    public long getReportedBy() {
        return reportedBy;
    }

    public void setReportedBy(long reportedBy) {
        this.reportedBy = reportedBy;
    }

    public Optional<Long> getAssignedTo() {
        return assignedTo;
    }

    public void setAssignedTo(Optional<Long> assignedTo) {
        this.assignedTo = assignedTo;
    }

    public BugStatus getStatus() {
        return status;
    }

    public void setStatus(BugStatus status) {
        this.status = status;
    }

}

File: src/main/scala/org/grimrose/sqlap22/domain/scala/Bug.scala
package org.grimrose.sqlap22.domain.scala

import org.joda.time.LocalDate
import scalikejdbc._

case class Bug(
                bugId: Long,
                dataReported: LocalDate,
                summary: Option[String] = None,
                reportedBy: Long,
                reportedAccount: Account,
                assignedTo: Option[Long] = None,
                assignedAccount: Option[Account] = None,
                status: String,
                bugStatus: BugStatus)

object Bug extends SQLSyntaxSupport[Bug] {
  override val tableName = "Bugs"

  def apply(b: SyntaxProvider[Bug], report: SyntaxProvider[Account], assign: SyntaxProvider[Account], bs: SyntaxProvider[BugStatus])(rs: WrappedResultSet): Bug =
    apply(b.resultName, report.resultName, assign.resultName, bs.resultName)(rs)

  def apply(b: ResultName[Bug], report: ResultName[Account], assign: ResultName[Account], bs: ResultName[BugStatus])(rs: WrappedResultSet): Bug =
    new Bug(
      bugId = rs.get(b.bugId),
      dataReported = rs.get(b.dataReported),
      summary = rs.get(b.summary),
      reportedBy = rs.get(b.reportedBy),
      reportedAccount = Account(rs.long(report.accountId)),
      assignedTo = rs.get(b.assignedTo),
      assignedAccount = Account.opt(assign)(rs),
      status = rs.get(b.status),
      bugStatus = BugStatus(rs.string(bs.status))
    )

}
